src/NServiceBus.AcceptanceTests/Routing/When_publishing_an_event_to_self.cs

```diff
@@ -0,0 +1,78 @@
+﻿namespace NServiceBus.AcceptanceTests.Routing
+{
+    using System;
+    using System.Threading.Tasks;
+    using AcceptanceTesting;
+    using EndpointTemplates;
+    using NUnit.Framework;
+
+    public class When_publishing_an_event_to_self : NServiceBusAcceptanceTest
+    {
+        [Test]
+        public async Task Event_should_be_received()
+        {
+            var ctx = await Scenario.Define<Context>(x => x.Id = Guid.NewGuid())
+                .WithEndpoint<PublisherAndSubscriber>(b => b
+                .When((session, context) =>
+                {
+                    if (context.HasNativePubSubSupport)
```

Comment:
since the check below is `c.EventSubscribed || c.HasNativePubSubSupport`, I think we can remove this when altogether?